packages/enzyme-test-suite/test/shared/hooks/useEffect.jsx

```diff
@@ -65,5 +66,192 @@ export default function describeUseEffect({
         done();
       }, timeout + 1);
     });
+
+    describe('with mount effect', () => {
+      const didMountCount = 9;
+
+      function FooCounterWithMountEffect({ initialCount = 0 }) {
+        const [count, setCount] = useState(+initialCount);
+
+        useEffect(() => {
+          setCount(didMountCount);
+        }, []);
+        return (
+          <Fragment>
+            <span className="counter">
+              {count}
+            </span>
+          </Fragment>
+        );
+      }
+
+      it('initial render after did mount effect', () => {
+        const wrapper = Wrap(<FooCounterWithMountEffect />);
+        expect(wrapper.find('.counter').text()).to.equal(String(didMountCount));
+      });
+    });
+
+    describe('with async effect', () => {
+      it('works with `useEffect`', (done) => {
+        const wrapper = Wrap(<ComponentUsingEffectHook />);
+
+        expect(wrapper.debug()).to.equal(
+          isShallow
+           ? `<div>
+  1
+</div>`
+          : `<ComponentUsingEffectHook>
+  <div>
+    1
+  </div>
+</ComponentUsingEffectHook>`,
+        );
+
+        setTimeout(() => {
+          wrapper.update();
+          expect(wrapper.debug()).to.equal(
+            isShallow
+           ? `<div>
+  2
+</div>`
+          : `<ComponentUsingEffectHook>
+  <div>
+    2
+  </div>
+</ComponentUsingEffectHook>`,
+          );
+          done();
+        }, timeout + 1);
+      });
+    });
+
+    describe('on componentDidUpdate & componentDidMount', () => {
+      const expectedCountString = x => `You clicked ${x} times`;
+
+      let setDocumentTitle;
+      function ClickCounterPage() {
+        const [count, setCount] = useState(0);
+
+        useEffect(() => {
+          setDocumentTitle(expectedCountString(count));
+        }, [count]);
+
+        return (
+          <div>
+            <p>You clicked {count} times</p>
+            <button onClick={() => setCount(count + 1)}>
+              Click me
+            </button>
+          </div>
+        );
+      }
+
+      beforeEach(() => {
+        setDocumentTitle = sinon.stub();
+      });
+
+      it('on mount initial render', () => {
+        const wrapper = Wrap(<ClickCounterPage />);
+
+        expect(wrapper.find('p').text()).to.eq(expectedCountString(0));
+        expect(setDocumentTitle).to.have.property('callCount', 1);
+        expect(setDocumentTitle.args).to.deep.equal([[expectedCountString(0)]]);
+      });
+
+      // TODO: useEffect fixme
```

Comment:
why is this broken on `mount`?

Reply:
Not broken, just need a `.act` wrap; I'd fix this with `.simulate` .